src/z3c/relationfield/relation.py

```diff
@@ -21,6 +23,10 @@ def __init__(self, to_id):
         self.__parent__ = None
         self.from_attribute = None
 
+    def __hash__(self):
+        """There should only be one RelationVaule per sort_key."""
+        return id(self) // 16
```

Comment:
I'm not sure it's safe to have persistent objects whose hashes change every time they're loaded from the ZODB.  I'd have to think about this more.

Reply:
Also, `id()` is very PyPy-unfriendly.  Maybe use `return object.__hash__(self)`?

Reply:
Fine for me!